compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -4324,6 +4328,139 @@ public void visit(BLangStatementExpression bLangStatementExpression) {
         result = bLangStatementExpression;
     }
 
+    @Override
+    public void visit(BLangQueryExpr queryExpr) {
+        List<BLangFromClause> fromClauseList = queryExpr.fromClauseList;
+        BLangFromClause fromClause = fromClauseList.get(0);
+        BLangSelectClause selectClause = queryExpr.selectClause;
+        List<BLangWhereClause> whereClauseList = queryExpr.whereClauseList;
+        DiagnosticPos pos = fromClause.pos;
+
+        // Create Foreach statement
+        //
+        // Below query expression :
+        //      from var person in personList
+        //
+        // changes as,
+        //      foreach var person in personList {
+        //          ....
+        //      }
+
+        BLangForeach leafForEach = null;
+        BLangForeach parentForEach = null;
+
+        for (BLangFromClause bLangFromClause : fromClauseList) {
+            BLangForeach foreach = (BLangForeach) TreeBuilder.createForeachNode();
+            foreach.pos = queryExpr.pos;
+            foreach.collection = bLangFromClause.collection;
+            types.setForeachTypedBindingPatternType(foreach);
+
+            foreach.variableDefinitionNode = bLangFromClause.variableDefinitionNode;
+            foreach.isDeclaredWithVar = true;
```

Comment:
Why do we set this field to true always?

Reply:
Fixed.